s3-get-files.sh

```diff
@@ -1,67 +1,146 @@
 #!/bin/bash
 
+# Set the endpoint URL for AWS CLI operations to handle Digital Ocean space.
+export AWS_ENDPOINT_URL=https://nyc3.digitaloceanspaces.com
+
 # Log file path
-log_file="/root/SCRIPTS/logfile.txt"
+log_file="/root/SCRIPTS/temp_logfile.txt"
 
 # Function to log messages with timestamp
 log() {
-    echo "$(date +"%Y-%m-%d %H:%M:%S") $1" >> "$log_file"
+        echo "$(date +"%Y-%m-%d %H:%M:%S") $1" >> "$log_file"
 }
 
+
+log "---------------------------------------------"
+
+# Setup the fixed-length processing directory name
 # Create a unique processing directory name with timestamp
-processing_dir="processing_$(date +"%Y-%m-%d_%H-%M-%S")"
-s3cmd mb "s3://infinity-91872/$processing_dir"
-log "Created processing directory: $processing_dir"
+static_prefix="processing_"
+base_processing_dir="$static_prefix$(date +"%Y-%m-%d_%H-%M-%S")"
+required_length=31
+padding_needed=$((required_length - ${#base_processing_dir}))
+padding=$(printf '%*s' "$padding_needed" '' | tr ' ' '_')
+processing_dir="${base_processing_dir}${padding}__"
+
+# Function to check if a file is a processing file
+is_processing_file() {
```

Comment:
`is_processing_file` function is for solving conflict when there are multiple requests to the certain file.